packages/@mantine/modals/src/events.ts

```diff
@@ -1,29 +1,8 @@
-import { createUseExternalEvents, DataAttributes } from '@mantine/core';
+import { createUseExternalEvents } from '@mantine/core';
 import { randomId } from '@mantine/hooks';
-import {
-  MantineModal,
-  MantineModals,
-  ModalSettings,
-  OpenConfirmModal,
-  OpenContextModal,
-} from './context';
+import { ModalsContextProps } from './context';
 
-type ModalsEvents = {
-  openModal: (payload: ModalSettings) => string;
-  openConfirmModal: (payload: OpenConfirmModal) => string;
-  openContextModal: <TKey extends MantineModal>(
-    payload: OpenContextModal<Parameters<MantineModals[TKey]>[0]['innerProps']> & {
-      modal: TKey;
-    } & DataAttributes
-  ) => string;
-  closeModal: (id: string) => void;
-  closeContextModal: <TKey extends MantineModal>(id: TKey) => void;
-  closeAllModals: () => void;
-  updateModal: (
-    payload: { modalId: string } & Partial<ModalSettings> & Partial<OpenConfirmModal>
-  ) => void;
-  updateContextModal: (payload: { modalId: string } & Partial<OpenContextModal<any>>) => void;
-};
+type ModalsEvents = Omit<ModalsContextProps, 'modals' | 'modalProps'>;
```

Comment:
I took the liberty of doing it like this but I can also revert it back to defining each function. Doing it like that is less error prone in my opinion as everything type related is actually concentrated in `Context.ts`.